src/components/Wizard/CreateVmWizard/providers/vmwareActions.js

```diff
@@ -0,0 +1,187 @@
+import { getImportProviderSecretObject, getDefaultSecretName, getV2VVMwareObject } from './vmwareProviderPod';
+
+import {
+  NAMESPACE_KEY,
+  PROVIDER_STATUS_CONNECTING,
+  PROVIDER_STATUS_CONNECTION_FAILED,
+  PROVIDER_VMWARE_URL_KEY,
+  PROVIDER_VMWARE_USER_NAME_KEY,
+  PROVIDER_VMWARE_USER_PWD_AND_CHECK_KEY,
+  PROVIDER_VMWARE_USER_PWD_KEY,
+  PROVIDER_VMWARE_CONNECTION,
+} from '../constants';
+import { CONNECT_TO_NEW_INSTANCE } from '../strings';
+import { settingsValue } from '../../../../k8s/selectors';
+import { getName } from '../../../../utils/selectors';
+
+export const onVmwareCheckConnection = async (basicSettings, onChange, k8sCreate) => {
+  // Note: any changes to the dialog since issuing the Check-button action till it's finish will be lost due to tight binding of the onFormChange to basicSettings set at promise creation
+  onChange({ status: PROVIDER_STATUS_CONNECTING });
+
+  const url = settingsValue(basicSettings, PROVIDER_VMWARE_URL_KEY);
+  const username = settingsValue(basicSettings, PROVIDER_VMWARE_USER_NAME_KEY);
+  const password = get(
+    settingsValue(basicSettings, PROVIDER_VMWARE_USER_PWD_AND_CHECK_KEY),
+    PROVIDER_VMWARE_USER_PWD_KEY
+  );
+  const namespace = settingsValue(basicSettings, NAMESPACE_KEY);
+
+  try {
+    const secretName = `temp-${getDefaultSecretName({ url, username })}-`;
+    const secret = await k8sCreate(
+      SecretModel,
+      getImportProviderSecretObject({
+        url,
+        username,
+        password,
+        namespace,
+        secretName,
+        isTemporary: true, // do not list this temporary secret in the dropdown box
+      })
+    );
+
+    const v2vVmware = await k8sCreate(
+      V2VVMwareModel,
+      getV2VVMwareObject({
+        name: `check-${getDefaultSecretName({ url, username })}-`,
+        namespace,
+        connectionSecretName: getName(secret),
+        isTemporary: true, // remove this object automatically (by controller)
+      })
+    );
+
+    onChange({ V2VVmwareName: getName(v2vVmware), status: PROVIDER_STATUS_CONNECTING }); // still "connecting" here, let content in the "status" of the CR decide otherwise (set by controller)
+  } catch (err) {
+    console.warn('onVmwareCheckConnection(): Check for VMWare credentials failed, reason: ', err); // eslint-disable-line no-console
+    onChange({ status: PROVIDER_STATUS_CONNECTION_FAILED }); // The CR can not be created
+  }
+};
+
+// secret already exists, use it
+export const onVCenterInstanceSelected = async (
+  k8sCreate,
+  valueValidationPair,
+  key,
+  formValid,
+  prevBasicSettings,
+  onFormChange
+) => {
+  // hint: onFormChange = (newValue, key, formValid) => {} to update basicSettings
+  const { value } = valueValidationPair;
+  const connectionSecretName = value;
+  if (!connectionSecretName || connectionSecretName === CONNECT_TO_NEW_INSTANCE) {
+    setTimeout(() => {
+      // let other events to be processed
+      onFormChange(
```

Comment:
why does this have to be postponed? Btw if by any chance the CreateVmDialog gets closed / unmounted, it will call setState on unmounted component.

Reply:
It's about how the onChange/onFormeChange functions bubbles-through the event handling, the value set here is later replaced by something else. Rearranging with just `setTimeout(..., 0)` works fine as well.

Reply:
I see, 0 looks much better now.